### PR TITLE
Water Tank QoL

### PR DESCRIPTION
**Pull request by PhantomEpicness**
## About The Pull Request

In yet another very funny story of "fixing" a problem illogically that simply creates more, we have watertanks.

Sometime in the past someone abused them to block xenos as they weren't destroyable by them.

So what did a dev do?

_Make them destructable?_

Nope, that's too simple. Instead, we go to the almayer and permanently anchor the watertanks there! Fixed!

Hilariously, not even all of them were anchored.

This PR makes so all almayer watertanks are unanchored as well as being slashable and not bulletproof.

Also unanchored the two custom OT tanks to the right of OT fuel room.

## Why It's Good For The Game

water tank consistency + qol + bugfix + less stupid shit that's bulletproof/slashproof
## Changelog

:cl:totalepicness

qol: Liquid tanks are now destroyable by bullets and xenos alike
qol: Water tanks can now be wrenched and unwrenched at will.
fix: Fixed the random anchored watertanks in Almayer
fix: Fixed anchored OT custom reagent tanks in OT fuel room

/:cl:

### CODE REVIEW

File: code/game/objects/structures/reagent_dispensers.dm [hunk 1]
```diff
@@ -54,6 +63,22 @@
...
 		amount_per_transfer_from_this = N
 
+/obj/structure/reagent_dispensers/proc/healthcheck()
+	if(health <= 0)
+		Destroy()
+
+/obj/structure/reagent_dispensers/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage
+	if(Proj.firer)
+		message_staff("[key_name_admin(Proj.firer)] fired a projectile at [name] in [loc.loc.name] ([loc.x],[loc.y],[loc.z]) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];adminplayerobservecoodjump=1;X=[loc.x];Y=[loc.y];Z=[loc.z]'>JMP</a>).")
```
Comment (stanalbatross) [inline code/game/objects/structures/reagent_dispensers.dm:73]: make this an attack log or niche log

File: code/game/objects/structures/reagent_dispensers.dm [hunk 2]
```diff
@@ -54,6 +63,22 @@
...
+		log_game("[key_name(Proj.firer)] fired a projectile at [name] in [loc.loc.name] ([loc.x],[loc.y],[loc.z]).")
+	playsound(src, 'sound/effects/metalhit.ogg', 25, 1)
+	if(health <= 0)
+		Destroy()
```
Comment (stanalbatross) [inline code/game/objects/structures/reagent_dispensers.dm:77]: you make a proc for healthcheck, use it

File: maps/map_files/USS_Almayer/USS_Almayer.dmm [hunk 3]
```diff
@@ -56458,7 +56456,7 @@
 	tag = "icon-tube1 (WEST)"
 	},
 /obj/structure/reagent_dispensers/watertank{
-	anchored = 1
+	anchored = 0
```
Comment (stanalbatross) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:56459]: remove instance just replace with base path like you did above

File: maps/map_files/USS_Almayer/USS_Almayer.dmm [hunk 4]
```diff
@@ -67742,7 +67740,7 @@
 	tag = "icon-N"
 	},
 /obj/structure/reagent_dispensers/watertank{
-	anchored = 1
+	anchored = 0
```
Comment (stanalbatross) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:67743]: ''

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by nuklearcellphoneg** [comment]
Thank you, this will save me the ten minutes of dragging the unanchored one upstairs downstairs

File: code/game/objects/structures/reagent_dispensers.dm [hunk 5]
```diff
@@ -6,8 +6,10 @@
 	icon_state = "watertank"
 	density = 1
 	anchored = 0
+	health = 100 // Can be destroyed in 2-4 slashes.
 	flags_atom = CAN_BE_SYRINGED
-
+	wrenchable = TRUE
+	var/slashable = FALSE
```
Comment (stanalbatross) [inline code/game/objects/structures/reagent_dispensers.dm:12]: you haven't even implemented attack_alien

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by Wintermote** [comment]
PR's like this are a godsend.

File: code/game/objects/structures/reagent_dispensers.dm (hunk 5 above)
Comment (stanalbatross) [inline code/game/objects/structures/reagent_dispensers.dm:12]: this ia a base var on structure do not make a snowflake one

File: code/game/objects/structures/reagent_dispensers.dm [hunk 6]
```diff
@@ -39,6 +41,13 @@
 	else
 		to_chat(user, SPAN_NOTICE(" Nothing."))
 
+/obj/structure/reagent_dispensers/Destroy()
+	playsound(src.loc, 'sound/effects/slosh.ogg', 50, 1, 3)
+	visible_message(SPAN_NOTICE("\The [src] falls apart as its contents spill everywhere!"))
+	. = ..()
+
+
+
```
Comment (stanalbatross) [inline code/game/objects/structures/reagent_dispensers.dm:50]:
```suggestion
```

File: code/game/objects/structures/reagent_dispensers.dm [hunk 7]
```diff
@@ -54,6 +63,29 @@
 	if(N)
 		amount_per_transfer_from_this = N
 
+/obj/structure/reagent_dispensers/proc/healthcheck()
+	if(health <= 0)
+		Destroy()
```
Comment (stanalbatross) [inline code/game/objects/structures/reagent_dispensers.dm:68]: qdel

File: code/game/objects/structures/reagent_dispensers.dm [hunk 8]
```diff
@@ -54,6 +63,29 @@
...
+	playsound(src, 'sound/effects/metalhit.ogg', 25, 1)
+	healthcheck()
+	return TRUE
+
+/obj/structure/reagent_dispensers/attack_alien(mob/living/carbon/Xenomorph/user)
+	user.animation_attack_on(src)
+	health -= (rand(user.melee_damage_lower, user.melee_damage_upper))
+	playsound(src, 'sound/effects/metalhit.ogg', 25, 1)
+	user.visible_message(SPAN_DANGER("[user] slashes [src]!"), \
+	SPAN_DANGER("You slash [src]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
```
Comment (stanalbatross) [inline code/game/objects/structures/reagent_dispensers.dm:84]: \the

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/reagent_dispensers.dm (hunk 7 above)
Comment (stanalbatross) [reply code/game/objects/structures/reagent_dispensers.dm:65]: You need to remove the destroy line

**Review by stanalbatross — CHANGES_REQUESTED** [review]
Do not call destroy

File: code/game/objects/structures/reagent_dispensers.dm (hunk 7 above)
Comment (PhantomEpicness) [reply code/game/objects/structures/reagent_dispensers.dm]: meant to remove it lol

**Comment by PhantomEpicness** [comment]
> Do not call destroy
meant to remove it lol

fixed

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**